tun/checksum.go

```diff
@@ -3,23 +3,99 @@ package tun
 import "encoding/binary"
 
 // TODO: Explore SIMD and/or other assembly optimizations.
+// TODO: Test native endian loads. See RFC 1071 section 2 part B.
 func checksumNoFold(b []byte, initial uint64) uint64 {
```

Comment:
Wondering - did you see any improvements in actual network throughput, or just in the microbenchmark? I ask because unrolling is often faster in isolation but sometimes being self-defeated when it blows the icache.

Reply:
I measured ~10% throughput improvement with unrolling on i5-12400 systems.

Also, @sailorfrag has a SIMD checksum PR in the works with SSE2 and AVX2 implementations that improves on this further.